### ISSUE TITLE
8chan incorrect "images" property

### ISSUE THREAD

**Issue by catamphetamine** [vichan-devel/vichan#327]
Hello.
I'm developing a [modern GUI](https://github.com/catamphetamine/chanchan) for imageboards and I've integrated `vichan` engine support in it.
https://catamphetamine.github.io/chanchan/?chan=8ch

While adding support for `vichan` engine I found a small bug: the `images` property of `catalog.json` is incorrect (is usually `0`, but I might have seen it `1` or something like that, though it's not certain).

For example, see the "Moderation" thread on 8ch.net:
https://8ch.net/leftypol/index.html

Screenshot:
![image](https://user-images.githubusercontent.com/477167/54666871-fb027600-4afb-11e9-8ac7-f5d85db315bd.png)

The API for `catalog.json` returns `images: 0`.

Screenshot:

![image](https://user-images.githubusercontent.com/477167/54666830-e7efa600-4afb-11e9-81ca-bb99417581e8.png)

The workaround I'm using displaying `omitted_images` instead of `images` and it seems to be ok.

The same bug is present on [kohlchan.net](https://kohlchan.net).
`kohlchan.net` is running on `vichan`, that's why I'm reporting the issue here.
I know that `8chan` is based of a fork of a fork of `vichan` called [OpenIB](https://github.com/OpenIB/OpenIB/) so I also reported the issue there.

**Comment by ctrlcctrlv** [comment]
What is it supposed to be?

**Comment by catamphetamine** [comment]
@ctrlcctrlv I don't know, you tell me.
According to the docs: https://github.com/vichan-devel/vichan-API

attribute | value | description | possible values | example value
-- | -- | -- | -- | --
images | integer | # images total | 0-99999 | 132

Clearly `0` is not `# images total` in the cases I described because `omitted_images` is over `100` still `images` is `0`.

**Comment by catamphetamine** [comment]
@ctrlcctrlv Here's a screenshot of `kohlchan.net` which uses `vichan` too:

![image](https://user-images.githubusercontent.com/477167/54738636-2356a700-4bc6-11e9-8b29-bee9cf65c1ce.png)

**Comment by catamphetamine** [comment]
To clarify myself: I'm not urging this bug to be fixed because chan owners won't update anyway.
So it's the maintainers' decision on whether they want it fixed or not.
It might as well stay not fixed because the engine itself is now deprecated.
But the issue should stay for other people coming from Google who will be developing apps based on the API.

**Comment by catamphetamine** [comment]
@ctrlcctrlv So I actually revised the README and ...
omg, Frederick Brennen, so you're the guy behind 8ch.net.
Wow, nice to meet you.
I knew I heard the name somewhere.

So, since you're not the original author of the engine it's no surprise you're not familiar with that `images` property too.
I'd say we leave it as is then.
Really, no need to spend time on this, it's not a "security problem or other serious bug".
Ignore it.
But leave the issue open so that people developing mobile apps would know.

**Comment by ctrlcctrlv** [comment]
Right...I'm not sure why this is happening to be honest with you, the original API Marcin made was heavily based on 4chan's API obviously for compatibility, than I hacked in multiple image support. Looks like you may be able to just count the number of elements in the `tim` parameter, making the field redundant in any case.

**Comment by ctrlcctrlv** [comment]
Bizarre, looks like we've diverged from 4chan's API, which also has an `images` parameter that does something else entirely and is marked "only display on OPs"&hellip;

https://github.com/4chan/4chan-API

**Comment by catamphetamine** [comment]
@ctrlcctrlv 
> Looks like you may be able to just count the number of elements in the tim parameter, making the field redundant in any case.

Yep, doing exactly that:
https://github.com/catamphetamine/chanchan/blob/ca178afe720216844fd0fe78569799c351319a30/src/chan-parser/4chan/parseThread.js#L101-L131

This workaround does work indeed and that's the reason this is a very minor bug not affecting the user experience at all.

**Comment by ctrlcctrlv** [comment]
Do you know what the field does on 4chan?

**Comment by catamphetamine** [comment]
@ctrlcctrlv I guess I do: `images` is the count of all images in a thread except `last_replies` and except the main post.
For example, see this thread: https://boards.4channel.org/x/thread/21676943

`catalog.json` response:

```js
[{
	"page": 1,
	"threads": [{
		"no": 21676943,
		"sticky": 1,
		"closed": 1,
		"now": "10/31/18(Wed)03:15:47",
		"name": "Anonymous",
		"com": "Welcome to /x/ - Paranormal. This is not a board for the faint of heart. If you need something to get started with, see the below lists for some basic resources. We hope you enjoy your venture into the spooks, the creeps and the unknown.\r\n<br>\r\n<br>\r\nThe resources in this thread are not exhaustive and are merely meant for beginners to get their footing.\r\n<br>\r\n<br>\r\n<img src=\"//s.4cdn.org/image/temp/danger.gif\" alt=\"\">",
		"filename": "x1",
		"ext": ".jpg",
		"w": 591,
		"h": 453,
		"tn_w": 250,
		"tn_h": 191,
		"tim": 1540970147550,
		"time": 1540970147,
		"md5": "p4mZMmoQme4FXqMryrlq5A==",
		"fsize": 39940,
		"resto": 0,
		"capcode": "mod",
		"semantic_url": "welcome-to-x-paranormal-this-is-not-a-board-for",
		"replies": 2,
		"images": 2,
		"omitted_posts": 1,
		"omitted_images": 1,
		"last_replies": [{
			"no": 21676989,
			"now": "10/31/18(Wed)03:28:59",
			"name": "Anonymous",
			"com": "Please note the following:<br><br>• This board desires high quality discussion. High quality posts will be praised. Low quality posts e.g. &quot;Is this paranormal?&quot; or &quot;I am [insert paranormal entity here] ask me anything,&quot; etc. will be removed.<br>• Requests belong on <a href=\"/wsr/\" class=\"quotelink\">&gt;&gt;&gt;/wsr/</a>.<br>• Conspiracy theories are welcome, but anything political in nature should be posted to <a href=\"/pol/\" class=\"quotelink\">&gt;&gt;&gt;/pol/</a>.<br>• For everything else, refer to global and board-specific rules.",
			"filename": "x3",
			"ext": ".jpg",
			"w": 540,
			"h": 533,
			"tn_w": 125,
			"tn_h": 123,
			"tim": 1540970939593,
			"time": 1540970939,
			"md5": "7167RjBOn1mYDR/uryf0wg==",
			"fsize": 58152,
			"resto": 21676943,
			"capcode": "mod"
		}],
		"last_modified": 1541003242
	}]
}]
```

`images` is `2` and `omitted_images` is `1` with `last_replies` having one attachment and the main post having one attachment.
The thread has 3 posts total with an attachment in each post.
The second post is not part of `last_replies` so it's "omitted".

**Comment by ctrlcctrlv** [comment]
OK I get it. I think if this were to be fixed it should be made to match 4chan's use. I'd accept a PR for that.
